src/lib/homestore_backend/hs_homeobject.hpp

```diff
@@ -81,6 +100,43 @@ class HSHomeObject : public HomeObjectImpl {
         static ShardInfo shard_info_from_sb(homestore::superblk< shard_info_superblk > const& sb);
     };
 
+#pragma pack(1)
+    // Every blob stored in disk as blob header | blob data | blob metadata(optional).
+    struct BlobHeader {
+        static constexpr uint64_t blob_max_hash_len = 32;
+        static constexpr uint8_t blob_header_version = 0x01;
+        static constexpr uint64_t blob_header_magic = 0x21fdffdba8d68fc6; // echo "BlobHeader" | md5sum
+
+        enum class HashAlgorithm : uint8_t {
+            NONE = 0,
+            CRC32 = 1,
+            MD5 = 2,
+        };
+
+        uint64_t magic{blob_header_magic};
+        uint8_t version{blob_header_version};
+        HashAlgorithm hash_algorithm;
+        uint8_t hash[blob_max_hash_len]{};
+        shard_id_t shard_id;
+        int64_t meta_data_offset{}; // Metadata stored after the blob data.
+        int64_t total_size{};       // Size of data and metadata.
```

Comment:
as we have implementation limit that sisl::io_blob_safe take uint32 as length as well as the `compute_blob_hash` takes `sisl::blob`  also capping to uint32,  maybe we can use uint32 for all the size and do a capping for put request? 

Though it is fair enough to cap the MAX_BLOB_SIZE to 1GB/2GB, no need to play at the boundary of uint32

Reply:
Ack